service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/WebClientFilterService.java

```diff
@@ -75,11 +80,29 @@ private static ExchangeFilterFunction errorHandling(RequestType requestType, Htt
             if (requestType.equals(RequestType.MHS_OUTBOUND) && clientResponse.statusCode().is5xxServerError()) {
                 return Mono.error(new MhsServerErrorException("MHS responded with status code " + clientResponse.statusCode().value()));
             }
+            if (requestType.equals(RequestType.MHS_OUTBOUND) && clientResponse.statusCode().value() == BAD_REQUEST.value()) {
+                return clientResponse.bodyToMono(String.class)
+                    .flatMap(WebClientFilterService::handle400FromMhsOutbound);
+            }
 
             return getResponseError(clientResponse, requestType);
         });
     }
 
+    private static Mono<ClientResponse> handle400FromMhsOutbound(String body) {
+
+        Pattern pattern = Pattern.compile(MAX_ATTACHMENTS_REGEX, CASE_INSENSITIVE);
```

Comment:
This way of checking what the response is feels super janky.

I'm not saying we should do this, but in the ideal world some sort of machine readable API response would be much more preferable from MHS.  I'm surprised that the MHS doesn't already provide this 😭

Reply:
Yeah, it would be better if the MHS adaptor had returned the validation errors as JSON. I was surprised it plain text too. I thought this was the best way to check it, given that it currently is.